packages/protocol/test/ACE/ACEAdjustSupply.js

```diff
@@ -0,0 +1,354 @@
+                erc20.address,
+                scalingFactor,
+                canAdjustSupply,
+                canConvert,
+                { from: accounts[0] }
+            );
+
+            const { receipt: mintReceipt } = await ace.mint(MINT_PROOF, proofs[0].proofData, accounts[0]);
+            expect(mintReceipt.status).to.equal(true);
+
+            const { receipt: aceReceipt } = await ace.validateProof(JOIN_SPLIT_PROOF, accounts[0], proofs[2].proofData);
+            expect(aceReceipt.status).to.equal(true);
+
+            const formattedProofOutput = `0x${proofOutput.slice(0x40)}`;
+            await truffleAssert.reverts(ace.updateNoteRegistry(JOIN_SPLIT_PROOF, accounts[0], formattedProofOutput));
+        });
+
+        it('validate failure if trying to convert value when the asset is NOT convertible', async () => {
+            const kPublic = 50; // kPublic is one greater than linked token balance
+            const tokensTransferred = new BN(50);
+            const publicOwner = accounts[1];
+            const inputNoteOwners = aztecAccounts.slice(2, 4);
+
+            proofs[2] = proof.joinSplit.encodeJoinSplitTransaction({
+                inputNotes: adjustedNotes,
+                outputNotes: [],
+                senderAddress: accounts[0],
+                inputNoteOwners,
+                publicOwner,
+                kPublic,
+                validatorAddress: aztecJoinSplit.address,
+            });
+
+            proofOutput = outputCoder.getProofOutput(proofs[2].expectedOutput, 0);
+
+            erc20 = await ERC20Mintable.new();
+            const scalingFactor = new BN(1);
+            const canAdjustSupply = true;
+            const canConvert = false;
+
+            await erc20.mint(
+                accounts[0],
+                scalingFactor.mul(tokensTransferred),
+                { from: accounts[0], gas: 4700000 }
+            );
+
+            await erc20.approve(
+                ace.address,
+                scalingFactor.mul(tokensTransferred),
+                { from: accounts[0], gas: 4700000 }
+            );
+
+            await ace.createNoteRegistry(
+                erc20.address,
+                scalingFactor,
+                canAdjustSupply,
+                canConvert,
+                { from: accounts[0] }
+            );
+
+            const { receipt: mintReceipt } = await ace.mint(MINT_PROOF, proofs[0].proofData, accounts[0]);
+            expect(mintReceipt.status).to.equal(true);
+
+            const { receipt: aceReceipt } = await ace.validateProof(JOIN_SPLIT_PROOF, accounts[0], proofs[2].proofData);
+            expect(aceReceipt.status).to.equal(true);
+
+            const formattedProofOutput = `0x${proofOutput.slice(0x40)}`;
+            await truffleAssert.reverts(ace.updateNoteRegistry(JOIN_SPLIT_PROOF, accounts[0], formattedProofOutput));
+        });
+
+        it('validate failure if user has not approved ACE to extract tokens', async () => {
```

Comment:
We should test that `mint` can't be called on an asset that is not mintable

Reply:
Moved to testing checklist